assets/src/components/site-scan-context-provider/index.js

```diff
@@ -59,12 +60,28 @@ const STATUS_CANCELLED = 'STATUS_CANCELLED';
  * @type {Object}
  */
 const INITIAL_STATE = {
-	currentlyScannedUrlIndex: 0,
+	currentlyScannedUrlIndexes: [],
 	forceStandardMode: false,
+	mostRecentlyScannedUrlIndex: -1,
```

Comment:
Is `mostRecentlyScannedUrlIndex` the best name? What if I'm scanning 3 URLs concurrently, and the 3rd finishes before the 1st. Would that be a problem? It doesn't seem so, as I tried scanning with this logic running:

```php
add_action( 'wp_head', function () {
	$sleep = rand( 0, 5 );
	sleep( $sleep );
} );
```

Reply:
I had issues with naming some variables in this PR. The `mostRecentlyScannedUrlIndex` is one of them and, as I think of it right now, it doesn't describe its purpose well.

Whenever a next URL (out of the `scannableUrls` array) is being validated, we store its index in this variable. It's used to calculate a current progress value and to pick a correct URL type/name to be displayed under the progress bar.

In 47cf600  I decided to go with `scannedUrlsMaxIndex`. It still feels to me a bit inaccurate but I wasn't able to come up with anything better. Feel free to change it to something else if you have any other suggestion.

In the same [commit](https://github.com/ampproject/amp-wp/pull/6705/commits/47cf6004e1da68687dc68f16c21c5921977828f2), I've also changed the way the value of the `scannedUrlsMaxIndex` is determined. Instead of keeping it in the state, it's now derived from the state. Its value is based on the queue of `urlIndexesPendingScan` and not on the `currentlyScannedUrlIndexes` array:

```diff
- const scannedUrlsMaxIndex = Math.max( ...currentlyScannedUrlIndexes );
+ const scannedUrlsMaxIndex = Math.min( scannableUrls.length, ...urlIndexesPendingScan ) - 1;
```

I took this approach to prevent potential weird bugs. Let's say that there are 8 scannable URLs. The 5th one is validated way longer than usual. All the remaining URLs get queued and validated while the 5th one is still waiting for a response. For the most time, the progress bar shows the correct value. However, in the end the `currentlyScannedUrlIndexes` array contains only the index of the 5th URL and the progress bar unexpectedly goes in the opposite direction.

By using the `urlIndexesPendingScan` we're sure that the progress bar never goes backwards.

Reply:
The name `scannedUrlsMaxIndex` works for me. Another option would be `highestScannedUrlIndex`.